🏗 Switch from google-camelcase to local eslint plugin

jridgewell opened the pull request:
This allows us to extend the exceptions and add `_Enum` as an approved case.

Partial for #36754

samouri commented:
Did you intentionally remove the package-lock.json file?

rsimha commented:
> ~1. Did you intentionally remove the package-lock.json file?~ edit: it isn't removed

The reason `package-lock.json` looked like it was removed was the large number of lines changed. This is almost certainly due to the use of an older (non-active-lts) version of `npm`. Best way to fix is to revert `package-lock.json`, run `nvm install --lts`, make sure you're on node 16 / npm 8, and re-run `npm install`.

jridgewell commented:
> 1. Did you intentionally remove the package-lock.json file? edit: it isn't removed

Raghu's answer is correct, I had an old terminal window open that still had node v14. Fixed.

> 2. I might be going crazy...but I can't find the camelcase rule in this PR. Did you forget to add the file?

Yup, forgot to commit it. Fixed.
